addons/safemode/functions/fn_safeWeapon.sqf

```diff
@@ -0,0 +1,33 @@
+/*
+ * Author: Brostrom.A
+ * Safe or unsafe the given weapon based on weapon state; locked or unlocked.
+ *
+ * Arguments:
+ * 0: Unit <OBJECT>
+ * 1: Weapon <STRING>
+ * 2: Muzzle <STRING>
+ * 3: State <BOOL>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [ACE_player, currentWeapon ACE_player, currentMuzzle Ace_player, true] call ace_safemode_fnc_safeWeapon
+ *
+ * Public: yes
+ */
+
+params [
+    ["_unit", objNull, [objNull]], 
+    ["_weapon", "", [""]],
+    ["_muzzle", "", [""]],
+    ["_state", true, [true]]
+];
+
+private _safedWeapons = _unit getVariable [QGVAR(safedWeapons), []];
+
+_weapon = configName (configFile >> "CfgWeapons" >> _weapon);
+
+if ((_state && _weapon in _safedWeapons) || (!_state && _weapon in _safedWeapons)) then {
```

Comment:
This logic doesn't work.

S: `_state`
W: `_weapon in _safedWeapons`

(S ∧ W) ∨ (¬S ∧ W)
W ∧ (S ∨ ¬S)
W ∧ T
W

So the function would only ever call `FUNC(lockSafety)` if W = `_weapon in _safedWeapons` is true: the function can only ever unlock a weapon.

I think the condition is just supposed to say:
`!(_state isEqualTo (_weapon in _safedWeapons))`

Reply:
it was sopuse to have been like this:
`((_state && _weapon in _safedWeapons) || (!_state && !(_weapon in _safedWeapons)))`
major typo on my part. 

whould your statement make the weapon locked if true and unlock in false?

Reply:
It would lock it if _state is true and it isn't already locked, and unlock it if _state is false and it is locked atm.

Reply:
I see the logic now weary nice!